app/models/release.rb

```diff
@@ -61,4 +62,14 @@ def contains_commit?(other_commit)
     Airbrake.notify(e, parameters: { github_repo: project.github_repo, commit: commit, other_commit: other_commit})
     false # Err on side of caution and cause a new release to be created.
   end
+
+  private
+
+  def covert_ref_to_sha
+    return if commit =~ Build::SHA1_REGEX
```

Comment:
If `commit` is blank, not worth updating the local cache below. I'd change this to

```
return if commit.nil? || commit =~ Build::SHA1_REGEX
```